### PR TITLE
[Backport2] Fix + sign HTTP folder

### PR DESCRIPTION
**Pull request by NomDuProfil**
## Description

Backport https://github.com/xbmc/xbmc/pull/16402

Fix wrong backport https://github.com/xbmc/xbmc/pull/16465

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/filesystem/HTTPDirectory.cpp [hunk 1]
```diff
@@ -103,6 +103,18 @@ bool CHTTPDirectory::GetDirectory(const CURL& url, CFileItemList &items)
         strLinkOptions = strLinkBase.substr(pos);
         strLinkBase.erase(pos);
       }
+
+      // encoding + and ; to URL encode if it is not already encoded by http server used on the remote server (example: Apache)
+      // more characters may be added here when required when required by certain http servers
+      pos = strLinkBase.find_first_of("+");
```
Comment (arnova) [inline xbmc/filesystem/HTTPDirectory.cpp]: Why did you remove the ; ?

Comment (valoulou) [reply xbmc/filesystem/HTTPDirectory.cpp]: I had done tests I just added it again

**Comment by arnova** [comment]
Thanks!
